Form/Type/BooleanType.php

```diff
@@ -50,12 +56,26 @@ public function setDefaultOptions(OptionsResolverInterface $resolver)
     public function configureOptions(OptionsResolver $resolver)
     {
         $resolver->setDefaults(array(
-            'translation_domain' => 'SonataCoreBundle',
             'choices'            => array(
                 self::TYPE_YES  => 'label_type_yes',
                 self::TYPE_NO   => 'label_type_no',
             ),
             'transform' => false,
+
+            // @deprecated Deprecated as of SonataCoreBundle 2.3.10, to be removed in 3.0.
+            'catalogue' => 'SonataCoreBundle',
+
+            // remove in SonataCoreBundle 3.0
+            'translation_domain' => function (Options $options) {
+                if ($options['catalogue']) {
+                    return $options['catalogue'];
+                }
+
+                return $options['translation_domain'];
+            },
+
+            // uncomment in SonataCoreBundle 3.0
+            // 'translation_domain' => 'SonataCoreBundle',
```

Comment:
Just a comment like 'Use directly translation_domain in 3.0' would be sufficient.